simulation/chart.py

```diff
@@ -150,120 +175,119 @@ class ChartWidget:
         plot.setDownsampling(
             ds=downsample, auto=autoDownsample, method=downsampleMethod
         )
-        self.__plots[id] = plot
+        self._plots[id] = plot
 
     def add_scatter_plot(
         self,
         id: str,
         name: str,
-        size: Optional[float] = None,
-        color: Optional[Color] = None,
-        symbol: Optional[SymbolType] = None,
-        border: Optional[PenConfig] = None,
-        autoDownsample: Optional[bool] = None,
-        downsample: Optional[int] = None,
-        downsampleMethod: Optional[DownsampleMethod] = None,
+        size: float = DEFAULT_SYMBOL_SIZE,
+        color: Color = DEFAULT_SYMBOL_COLOR,
+        symbol: SymbolType = DEFAULT_SYMBOL_MARKER,
+        border: PenConfig = DEFAULT_SYMBOL_PEN,
+        autoDownsample: bool = DEFAULT_DOWNSAMMPLE_AUTO,
+        downsample: int = DEFAULT_DOWNSAMPLE_AMOUNT,
+        downsampleMethod: DownsampleMethod = DEFAULT_DOWNSAMPLE_METHOD,
     ):
         """Add a scatter plot to the chart.
 
         Args:
             id (str): Unique id of the plot in the chart
             name (str): Name (legend title) of the plot
-            size (float, optional): Size of the marker symbol. Defaults to None.
-            color (Color, optional): Color to fill the marker symbol. Defaults to None.
-            symbol (SymbolType, optional): Shape of the marker symbol. Defaults to None.
-            border (PenConfig, optional): Pen to draw border around the marker symbol. Defaults to None.
-            autoDownsample (bool, optional): If True, resample the data before plotting to avoid plotting multiple line segments per pixel. Defaults to None.
-            downsample (int,optional): Reduce the number of samples displayed by the given factor. Defaults  to None.
-            downsampleMethod (DownsampleMethod,optional): Method to use for downsampling. Defaults to None.
+            size (float): Size of the marker symbol. Defaults to `10.0`.
+            color (Color): Color to fill the marker symbol. Defaults to `RGBAColor(50,50,150)`.
+            symbol (SymbolType): Shape of the marker symbol. Defaults to `o`.
+            border (PenConfig): Pen to draw border around the marker symbol. Defaults to `PenConfig(color=RGBAColor(200,200,200))`.
+            autoDownsample (bool): If True, resample the data before plotting to avoid plotting multiple line segments per pixel. Defaults to `False`.
+            downsample (int): Reduce the number of samples displayed by the given factor.To disable, set to `1`. Defaults  to `1`.
+            downsampleMethod (DownsampleMethod): Method to use for downsampling. Defaults to `peak`.
         """
         config = {"name": name, "pen": None}
 
         if size is not None:
             config["symbolSize"] = size
 
-        if color is not None:
-            config["symbolBrush"] = _make_color(color)
+        config["symbolBrush"] = _make_color(color)
 
         if symbol is not None:
```

Comment:
These other RTTI checks on `None` look like they can go as well. For example, this check on `symbol`, as well as the above check on `size` and the below check on `border`.

Reply:
https://github.com/semiotic-ai/autoagora-agents/pull/21/commits/0cec361a58e9735babbfda7c7a0365c9e21579a2